### PR TITLE
Watchlist & Favorites

### PR DESCRIPTION
**Pull request by codingknite**

### CODE REVIEW

File: src/components/MovieCards.jsx [hunk 1]
```diff
@@ -2,92 +2,93 @@ import React, { useState } from "react";
...
-
+    .movie-container {
+      display: flex;
+      flex-wrap: wrap;
+      
       .movie-card {
-      width: 16.1%;
-      border-radius: 2px;
-      margin: 5px;
+        width: 16.1%;
```
Comment (idiglove) [inline src/components/MovieCards.jsx:16]: cant we just make 16%? haha

File: src/pages/Genres.jsx [hunk 2]
```diff
@@ -33,6 +36,10 @@ export default function BrowseMovies({ state, dispatch }) {
       )
   );
 
+  useEffect(() => {
+    dispatch({ type: "initState" });
+  }, [genreId, dispatch]);
+
```
Comment (idiglove) [inline src/pages/Genres.jsx:42]: can you explain whats happening here?

File: src/reducers/markedReducer.js [hunk 3]
```diff
@@ -0,0 +1,21 @@
+export default function markedMovies(marked, action) {
+    switch (action.type) {
+        case "bookmark":
+            if (marked.includes(action.payload)) {
+                const removedId = marked.filter((movie) => movie !== action.payload);
+                return [...removedId];
+            } else {
+                return [...marked, action.payload];
+            }
```
Comment (idiglove) [inline src/reducers/markedReducer.js:9]: can be simplified to: 

if (marked.includes(action.payload)) {
                const removedId = marked.filter((movie) => movie !== action.payload);
                return [...removedId];
}

 return [...marked, action.payload];

File: src/reducers/markedReducer.js [hunk 4]
```diff
@@ -0,0 +1,21 @@
...
+        case "watchlist":
+            if (marked.includes(action.payload)) {
+                const removedId = marked.filter((movie) => movie !== action.payload);
+                return [...removedId];
+            } else {
+                return [...marked, action.payload];
+            }
```
Comment (idiglove) [inline src/reducers/markedReducer.js:16]: same here

**Review by idiglove — CHANGES_REQUESTED** [review]
Nice, i have some comments tho

File: src/components/MovieCards.jsx (hunk 1 above)
Comment (codingknite) [reply src/components/MovieCards.jsx:16]: Lol Okay haha

File: src/pages/Genres.jsx (hunk 2 above)
Comment (codingknite) [reply src/pages/Genres.jsx:42]: In the reducer "initState" initializes the state the 1 so when the genreId on a page changes...the API call starts from page 1.
Before, the genreId would change and the page still loads from the current page number in the state

**Comment by codingknite** [comment]
> 
> 
> Nice, i have some comments tho

Nice. Bring em on

File: src/pages/Genres.jsx (hunk 2 above)
Comment (idiglove) [reply src/pages/Genres.jsx:42]: isnt it good UX to stay at current page?

Comment (codingknite) [reply src/pages/Genres.jsx:42]: I don't think so...Imagine someone was browsing through animated movies and they went up to page 10 but then decided to checkout scifi movies....the API call would pull up page 11....see now they just missed 10 pages 😄

Comment (idiglove) [reply src/pages/Genres.jsx:42]: hmm i see, i might be missing something, but sounds good then

Comment (codingknite) [reply src/pages/Genres.jsx:42]: Let me show you a demo.
Also, what other comments did you have?

Comment (idiglove) [reply src/pages/Genres.jsx:42]: look under Conversation tab

Comment (codingknite) [reply src/pages/Genres.jsx:42]: I'm sorry but is this not the conversation tab?

Comment (idiglove) [reply src/pages/Genres.jsx:42]: Screen Shot 2021-04-02 at 3 37 14 PM
look at all the conversations here

Comment (codingknite) [reply src/pages/Genres.jsx:42]: shot

I did. I was alluding to what you said here. I thought you had other comments besides everything you have said so far.

**Comment by idiglove** [comment]
@codingknite  
Screen Shot 2021-04-02 at 3 49 33 PM
clearly u didnt look hard enough ;P

**Comment by codingknite** [comment]
Lol...I saw those...I just didn't react to them. Just went ahead and worked on them. i guess thats it then. sorry po :)

**Comment by codingknite** [comment]
I've worked on this project for way too long lol 😄 Def finishing it this week

File: src/components/MovieCards.jsx (hunk 1 above)
Comment (idiglove) [reply src/components/MovieCards.jsx:16]: this is still the same..

File: src/components/NavBar.jsx [hunk 5]
```diff
@@ -52,13 +55,22 @@ export default function HomepageNav() {
             </div>
 
             <div>
-              <BsIcons.BsSearch />
+              <div className="saves">
+                <NavLink to={`/favorites`} className="link">
+                  Favorites
+                  {favorites.length > 0 ? ` (${favorites.length})` : null}
```
Comment (idiglove) [inline src/components/NavBar.jsx:61]: just do `{favorites.length > 0 && ` (${favorites.length})`}`

**Review by idiglove — CHANGES_REQUESTED** [review]
few changes :)

**Pull request merged**